### ISSUE TITLE
[BUG] 2.7.2-rc7- monitoring/grafana - when users click some grafana links, a 404 is thrown. 

### ISSUE THREAD

**Issue by AndrewHoffmanQA** [rancher/rancher#41036]
**Rancher Server Setup**
rancher 2.7.2-rc7
k8- v1.25.6
3 instances running etcd, worker and cp
bug testing done on local cluster
monitoring- 102.0.1+up40.1.2

**Describe the bug**
When the user clicks on a Grafana link they are redirected to a 404 page, this happens in multiple places within the UI. 
note : it doesnt happen in every place Grafana is displayed.
note: this is only happening on the local cluster

**To Reproduce**

**scenario 1**
1. install monitoring
2. navigation- side menu > monitoring 
3. scroll up to top of page and click on the grafana link 

**scenario 2**
1. on local cluster, install monitoring
2. navigate to cluster dashboard, scroll down to the area of the page where grafana metrics are displayed.
3. see embedded grafana dashboard not displayed

**scenario 3**
1. install monitoring
2. install prom federator
4. navigation - side menu > monitoring > project monitoring
5. create a project monitor
6. once green and active, click on the project monitor name (user is taken to proj monitor page)
7. click the grafana link between Prometheus and alertmanager

**scenario 4**
1. install monitoring
2. install prom federator
3. navigation - side menu > monitoring > project monitoring
5. create a project monitor
6. once green and active, click on the project monitor name (user is taken to proj monitor page)
7. the embedded grafana is not displayed correctly- throws 404

**Result**
user gets a 404 error

**Expected Result**
user sees the appropriate grafana metrics 

**Screenshots**

![image](https://user-images.githubusercontent.com/89030218/229188895-ebf6e0c6-26e7-488d-a967-757138c6bce5.png)
![image](https://user-images.githubusercontent.com/89030218/229189844-bc6e3675-0803-4a91-8fb8-d898d08b1912.png)
screenshot of error page 
![image](https://user-images.githubusercontent.com/89030218/229206956-09999780-7254-4564-bc4c-2fe4851b89aa.png)

**Additional context**
this new issue was generated from this backport [bug ](https://github.com/rancher/dashboard/issues/8380), upon further testing it was found that the grafana link scoped for fixing in the ticket was a dashboard issue, however the issues with these grafana links appear to be a backend issue. there is some investigation discussion in the ticket that may be helpful for dev resources. 

note: i confirmed this is new, was not able to reproduce in 2.7.1

note: some links are not broken, for example if you navigate to the cluster dashboard and scroll down, the grafana links on that page are working.

**Comment by MKlimuszka** [comment]
Due to the added QA work and the severity of this ticket. We've removed the release blocker label. But we will add a work around here and get it fixed in Q2

**Comment by prachidamle** [comment]
This issue seems to be appearing only in the local cluster. Monitoring and grafana installed in a downstream cluster do not face this 404 error.

**Comment by skanakal** [comment]
I was facing this issue only with **rancher-monitoring:100.2.0+up40.1.2** on local cluster running with rancher 2.6.11 and 2.7.1.

observation: with rancher 2.7.2, After installing the monitoring 40.1.2, 

Navigating to the Cluster home > Scroll to the bottom > Cluster Metrics, Kubernetes Component Metrics and Etcd Metrics are actually displaying the metrics unlike 2.6.11 & 2.7.1 ...

but   navigation- side menu > monitoring > scroll up to top of page and click on the grafana link  is showing "Page not found 404 Error"

**Comment by clemenko** [comment]
Any ETA on the fix?

**Comment by maxaudron** [comment]
The problem for us was the [`global.cattle.clusterId`](https://github.com/rancher/charts/blob/09015c1f0412e09adc92ac5adb4b65271d450c48/charts/rancher-monitoring/102.0.0%2Bup40.1.2/charts/grafana/templates/nginx-config.yaml#LL77C79-L77C104) field in the helm values not being set, as we are deploying the helm chart manually.

This causes the nginx proxy container to rewrite requests incorrectly, and in turn grafana not recognizing the path, leading to the 404

**Comment by AndrewHoffmanQA** [comment]
retest for this was requested in standup. retest failed. moving the bug to 'reopened' 

tested on 
local cluster
rancher image 6c48ba77-head
monitoring chart - 102.0.1+up40.1.2

**Comment by AndrewHoffmanQA** [comment]
**BUG SCOPE INCREASE**

repo steps
1. on local cluster, install monitoring
2. navigate to cluster dashboard 
the grafana section does not display metrics. instead throws a 404. 

note: only impacts local

cluster info-
- local env used
-HA, 3 nodes all roles
-rancher image 6c48ba77-head
-monitoring chart - 102.0.1+up40.1.2


![image.png](https://zube.io/files/rancher/b8bd6ba662b30bc011d1f1547d3dab64-image.png)

updating the body of the bug to reflect this scope increase. 

cc @geethub97 if you would prefer this new scope documented in a new ticket, please let me know

**Comment by nflynt** [comment]
The relevant dashboard ticket for local clusters is https://github.com/rancher/dashboard/issues/8885

Basically there was confusion between the expected proxy configuration on the backend and the URL generation to access the appropriate proxy on the frontend. Once UI has completed their work, testing is as follows:

- Install the appropriate Rancher version
- Install each Monitoring chart version in turn
- Access the Cluster view and click through the monitoring tabs. The correct embedded view should display; if we see either a 404 error **or** the grafana homepage, something has gone wrong
- Access the Monitoring tab and click the Grafana link. This should open in a new tab displaying the Grafana homepage, this will look slightly different depending on chart version. A 404 here indicates an error.

I'm currently running a quick investigation on downstream clusters to confirm that no backend changes are needed in that regard, and will update again with my findings when that's completed.

**Comment by prachidamle** [comment]
@AndrewHoffmanQA @ronhorton  At this point, there is a dashboard fix needed to test if this is working and there is no backend fix needed. Moving this ticket to-test but it will be blocked on https://github.com/rancher/dashboard/issues/8885

**Comment by prachidamle** [comment]
Since the UI ticket is moved to the backlog, moving this to Q3 milestone

**Comment by AndrewHoffmanQA** [comment]
moving this ticket to 're-opened'

testing - mixed results. passing back to dev. 

cluster information
tested on local cluster
rancher image-v2.7.5-rc4 
rancher-monitoring:102.0.1+up40.1.2 
prometheus-federator:2.1.0+up0.3.0 

scenario 1 - passed
scenario 2- failed
scenario 3-  failed

**Comment by AndrewHoffmanQA** [comment]
https://zube.io/files/rancher/e210596fc027acf76d57d59c189d9b75-2023-06-08_07-20-47.mp4

**Comment by xmolitann** [comment]
Hello Rancher team, any update on this issue?

**Comment by MKlimuszka** [comment]
UI, please review the screen recording Geet has posted to see if the behavior is acceptable to you.

**Comment by AndrewHoffmanQA** [comment]
https://github.com/rancher/rancher/assets/89030218/d72455fe-3f46-4438-93d7-c545ecd99a66

confirmed no 404s being thrown in image 2.7.5 on local or downstream

**Comment by AndrewHoffmanQA** [comment]
moving this to 'blocked' since it is now in UI review

**Comment by aceeric** [comment]
Folks - this set of steps reliably reproduces the 404 error with:

1. Rancher version: v2.7.5
2. Cluster type: local
3. Monitoring version: 102.0.1+up40.1.2

These steps start from a freshly provisioned CentOS 7.9 instance in AWS:
```
-bash-4.2$ cat /etc/centos-release
CentOS Linux release 7.9.2009 (Core)
```

Install Helm, RKE2, Cert Manager, and Rancher:
```
HELMVER=v3.11.3 && \
curl -sfL https://get.helm.sh/helm-$HELMVER-linux-386.tar.gz\
 -o /tmp/helm-$HELMVER-linux-386.tar.gz &&\
 tar -zxvf /tmp/helm-$HELMVER-linux-386.tar.gz -C /tmp &&\
 mv /tmp/linux-386/helm /usr/sbin/helm &&\
 rm -rf /tmp/helm-$HELMVER-linux-386.tar.gz /tmp/linux-386

export INSTALL_RKE2_VERSION=v1.25.11+rke2r1 && curl -sfL https://get.rke2.io | sh -

systemctl enable --now rke2-server.service &

while ! stat /etc/rancher/rke2/rke2.yaml &>/dev/null ; do sleep 10s; done

mkdir -p ~/.kube && cp /etc/rancher/rke2/rke2.yaml ~/.kube/config &&\
 cp /var/lib/rancher/rke2/bin/kubectl /usr/bin/kubectl

kubectl get po -A

helm repo add cert-manager https://charts.jetstack.io
helm install --namespace cert-manager --create-namespace cert-manager cert-manager/cert-manager --version 1.11.0 --set installCRDs=true

helm repo add rancher-stable https://releases.rancher.com/server-charts/stable
helm install --namespace cattle-system --create-namespace rancher rancher-stable/rancher --version 2.7.5 --set hostname=$(hostname)
```

Navigate to the local cluster in the Rancher UI and verify no "monitoring" left-side link exists.

Install Rancher Monitoring with CRDs:
```
curl -L https://github.com/rancher/charts/archive/refs/heads/dev-v2.7.tar.gz\
 -o /tmp/rancher-monitoring-dev-v2.7.tar.gz

mkdir -p /tmp/rancher-charts &&\
 tar -zxf /tmp/rancher-monitoring-dev-v2.7.tar.gz -C /tmp/rancher-charts

helm install\
 --namespace cattle-monitoring-system\
 --create-namespace\
 rancher-monitoring-crd\
 /tmp/rancher-charts/charts-dev-v2.7/charts/rancher-monitoring-crd/102.0.1+up40.1.2

helm install --namespace cattle-monitoring-system\
 rancher-monitoring\
 /tmp/rancher-charts/charts-dev-v2.7/charts/rancher-monitoring/102.0.1+up40.1.2
```

Refresh the Rancher UI and verify "monitoring" now appears as a left-side link.

Click left side "monitoring" then on the right click "Grafana"

Observe 404.

**Comment by bashofmann** [comment]
@aceeric When installing rancher-monitoring with the helm cli, you have to pass the Rancher URL as a value. The Grafana UI needs to know the hostname, otherwise it will not generate links and references to JS and CSS files correctly. Something like this:

```
grafana:
  global:
    cattle:
      clusterName: susecon-demo
      clusterId: c-m-x7w6shvs
      url: https://rancher.susecon.plgrnd.be

global:
  cattle:
    clusterName: susecon-demo
    clusterId: c-m-x7w6shvs
    url: https://rancher.susecon.plgrnd.be
```

The UI does this automatically.

**Comment by LukaszWasko** [comment]
> @aceeric When installing rancher-monitoring with the helm cli, you have to pass the Rancher URL as a value. The Grafana UI needs to know the hostname, otherwise it will not generate links and references to JS and CSS files correctly. Something like this:
> 
> ```
> grafana:
>   global:
>     cattle:
>       clusterName: susecon-demo
>       clusterId: c-m-x7w6shvs
>       url: https://rancher.susecon.plgrnd.be
> 
> global:
>   cattle:
>     clusterName: susecon-demo
>     clusterId: c-m-x7w6shvs
>     url: https://rancher.susecon.plgrnd.be
> ```
> 
> The UI does this automatically.

This is something new. I just tried update helm chart **with** and **without** your code and **I don't see any difference** between grafana URLs. In both cases I see 404 error (in Grafana UI) after clicking in Rancher Monitoring -> Grafana. Then when I click Dashboards it's just fine.
When I trying to open dashboard in new tab in my browser, the Rancher throws 404 error - it's so frustrating that I'm closed in one tab with Grafana.
Yes, I verified that helm yaml from rancher contains new values.

**Comment by prachidamle** [comment]
>> https://github.com/rancher/rancher/issues/41036#issuecomment-1638429784
@AndrewHoffmanQA Is there a UI ticket we are waiting this on? 
cc @gaktive @MKlimuszka

**Comment by gaktive** [comment]
@prachidamle as far as I understand, there's no UI ticket. Do keep me in the loop if something needs to be done in the dashboard but as noted earlier in this ticket, this looks to be purely backend.

**Comment by ronhorton** [comment]
**Pass** Verified each documented scenario functioned as expected in rancher `v2.7.6-rc2`
rancher-monitoring:102.0.1+up40.1.2
prometheus-federator:3.0.0+up0.3.3
![2023-08-07_16-02-15.png](https://zube.io/files/rancher/11de8a0de8f556781b6237d61ce337df-2023-08-07_16-02-15.png)
![2023-08-07_16-11-14.png](https://zube.io/files/rancher/f137b4b408ce57e09fc4b5f2d3c51b80-2023-08-07_16-11-14.png)
![2023-08-07_16-14-49.png](https://zube.io/files/rancher/35e03fcb480e4ea50b7f28a1d522e5ce-2023-08-07_16-14-49.png)
![2023-08-07_16-18-04.png](https://zube.io/files/rancher/7e84e24f10118eba6684d295ee44f290-2023-08-07_16-18-04.png)

**Comment by skanakal** [comment]
@erSitzt can you check this workaround https://github.com/rancher/rancher/issues/41036#issuecomment-1644240096 , its working for me...

**Comment by erSitzt** [comment]
@skanakal thanks, but i know how to access the monitoring. the cluster name is missing in the grafana nginx config. but i dont want to have manual steps when deploying lots of clusters :)

<img width="1138" alt="image" src="https://github.com/rancher/rancher/assets/1625376/dba55101-58c5-48e7-aee8-0eb1fd21f4f1">


This is a bug that did not exist in the previous version of the monitoring chart and should be fixed

**Comment by cukal** [comment]
Finding the clusterId: 

`curl -s "https://${RANCHER_SERVER}/v3/clusters?name=${CLUSTER_NAME}" -H 'content-type: application/json' -H "Authorization: Bearer $APITOKEN" --insecure |jq -r .data[].id`

From: [https://www.reddit.com/r/rancher/comments/gfo44t/get_id_of_existing_cluster_via_api/ ](https://www.reddit.com/r/rancher/comments/gfo44t/get_id_of_existing_cluster_via_api/)
